edsdme/scripts/utils.js

```diff
@@ -150,6 +150,73 @@ export function redirectLoggedinPartner() {
   window.location.assign(target);
 }
 
+export async function getRenewBanner() {
+  const programType = getCurrentProgramType();
+  const accountExpiration = getPartnerDataCookieValue(programType, 'accountexpiration');
+  if (!accountExpiration) return;
+
+  const expirationDate = new Date(accountExpiration);
+  const now = new Date();
+
+  let metadataKey;
+  let daysNum;
+
+  const differenceInMilliseconds = expirationDate - now;
+  const differenceInDays = Math.abs(differenceInMilliseconds) / (1000 * 60 * 60 * 24);
+  const differenceInDaysRounded = Math.floor(differenceInDays);
+
+  if (differenceInMilliseconds > 0) {
+    if (differenceInDaysRounded === 0) {
+      metadataKey = 'banner-account-expires-today';
+    } else if (differenceInDays < 30) {
+      metadataKey = 'banner-account-expires';
+      daysNum = differenceInDaysRounded;
+    } else {
+      return;
+    }
+  } else if (differenceInMilliseconds < 0) {
+    if (differenceInDays < 90) {
```

Comment:
I guess this is something to be properly groomed: after the 90 days, the banner will disappear? does not really make sense, but the user should not be able to sign after 90 days of being suspended

Reply:
@Ben-Zahler Right, after 90 days banner will disappear